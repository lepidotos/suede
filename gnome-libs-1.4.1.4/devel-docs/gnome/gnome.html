<HTML
><HEAD
><TITLE
>Gnome Library Reference Manual</TITLE
><META
NAME="GENERATOR"
CONTENT="Modular DocBook HTML Stylesheet Version 1.74b"><LINK
REL="NEXT"
TITLE="Gnome Library"
HREF="libgnome.html"></HEAD
><BODY
CLASS="BOOK"
BGCOLOR="#FFFFFF"
TEXT="#000000"
LINK="#0000FF"
VLINK="#840084"
ALINK="#0000FF"
><DIV
CLASS="BOOK"
><DIV
CLASS="TITLEPAGE"
><TABLE
WIDTH="100%"
BORDER="0"
BGCOLOR="#000000"
CELLPADDING="1"
CELLSPACING="0"
><TR
><TH
ALIGN="center"
VALIGN="center"
><FONT
COLOR="#FFFFFF"
SIZE="7"
><P
CLASS="TITLE"
><A
NAME="AEN2"
>Gnome Library Reference Manual</A
></P
></FONT
></TH
></TR
></TABLE
></DIV
><DIV
CLASS="TOC"
><DL
><DT
><B
>Table of Contents</B
></DT
><DT
><A
HREF="libgnome.html"
>Gnome Library</A
></DT
><DD
><DL
><DT
><A
HREF="gnome-gnome-config.html"
>gnome-config</A
>&nbsp;--&nbsp;Simple access to configuration values (50% done)</DT
><DT
><A
HREF="gnome-gnome-defs.html"
>gnome-defs</A
>&nbsp;--&nbsp;GNOME definitions for C++ linking.</DT
><DT
><A
HREF="gnome-gnome-dentry.html"
>gnome-dentry</A
>&nbsp;--&nbsp;Manipulation of .desktop files.</DT
><DT
><A
HREF="gnome-gnome-exec.html"
>gnome-exec</A
>&nbsp;--&nbsp;Execution of programs from GNOME applications</DT
><DT
><A
HREF="gnome-gnome-fileconvert.html"
>gnome-fileconvert</A
>&nbsp;--&nbsp;File format conversion functions.</DT
><DT
><A
HREF="gnome-gnome-help.html"
>gnome-help</A
>&nbsp;--&nbsp;Routines for displaying help</DT
><DT
><A
HREF="gnome-gnome-history.html"
>gnome-history</A
>&nbsp;--&nbsp;Keeping track of recently used documents.</DT
><DT
><A
HREF="gnome-gnome-i18n.html"
>gnome-i18n</A
>&nbsp;--&nbsp;Support for localization and internationalization.</DT
><DT
><A
HREF="gnome-gnome-metadata.html"
>gnome-metadata</A
>&nbsp;--&nbsp;File metadata information storage.</DT
><DT
><A
HREF="gnome-gnome-mime-info.html"
>gnome-mime-info</A
>&nbsp;--&nbsp;Routines to get information bound to a MIME type.</DT
><DT
><A
HREF="gnome-gnome-mime.html"
>gnome-mime</A
>&nbsp;--&nbsp;Routines to determine the MIME type of a file</DT
><DT
><A
HREF="gnome-gnome-paper.html"
>gnome-paper</A
>&nbsp;--&nbsp;Paper dimensions, printing unit conversions.</DT
><DT
><A
HREF="gnome-gnome-popt.html"
>gnome-popt</A
>&nbsp;--&nbsp;Command line argument parser</DT
><DT
><A
HREF="gnome-gnome-regex.html"
>gnome-regex</A
>&nbsp;--&nbsp;Regular expression cache implementation</DT
><DT
><A
HREF="gnome-gnome-remote.html"
>gnome-remote</A
>&nbsp;--&nbsp;User preference routines for executing commands on a remote host</DT
><DT
><A
HREF="gnome-gnome-score.html"
>gnome-score</A
>&nbsp;--&nbsp;Tracking score for games</DT
><DT
><A
HREF="gnome-gnome-sound.html"
>gnome-sound</A
>&nbsp;--&nbsp;Sound playing routines for GNOME applications</DT
><DT
><A
HREF="gnome-gnome-triggers.html"
>gnome-triggers</A
>&nbsp;--&nbsp;Hierarchical signal mechanism for application events.</DT
><DT
><A
HREF="gnome-gnome-url.html"
>gnome-url</A
>&nbsp;--&nbsp;Launch viewers for documents based on their URL</DT
><DT
><A
HREF="gnome-gnome-util.html"
>gnome-util</A
>&nbsp;--&nbsp;Miscellaneous utility functions for GNOME and C programs.</DT
><DT
><A
HREF="gnome-libgnome.html"
>libgnome</A
>&nbsp;--&nbsp;Non GUI part of the GNOME library.</DT
></DL
></DD
></DL
></DIV
></DIV
><DIV
CLASS="NAVFOOTER"
><BR
CLEAR="all"><BR><TABLE
WIDTH="100%"
BORDER="0"
BGCOLOR="#000000"
CELLPADDING="1"
CELLSPACING="0"
><TR
><TD
WIDTH="25%"
BGCOLOR="#C00000"
ALIGN="left"
>&nbsp;</TD
><TD
WIDTH="25%"
BGCOLOR="#0000C0"
ALIGN="center"
><FONT
COLOR="#FFFFFF"
SIZE="3"
><B
>&nbsp;</B
></FONT
></TD
><TD
WIDTH="25%"
BGCOLOR="#00C000"
ALIGN="center"
><FONT
COLOR="#FFFFFF"
SIZE="3"
><B
>&nbsp;</B
></FONT
></TD
><TD
WIDTH="25%"
BGCOLOR="#C00000"
ALIGN="right"
><A
HREF="libgnome.html"
><FONT
COLOR="#FFFFFF"
SIZE="3"
><B
>Next Page &#62;&#62;&#62;</B
></FONT
></A
></TD
></TR
><TR
><TD
COLSPAN="2"
ALIGN="left"
>&nbsp;</TD
><TD
COLSPAN="2"
ALIGN="right"
><FONT
COLOR="#FFFFFF"
SIZE="3"
><B
>Gnome Library</B
></FONT
></TD
></TR
></TABLE
></DIV
></BODY
></HTML
>