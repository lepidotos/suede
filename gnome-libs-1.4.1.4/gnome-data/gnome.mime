#
#	NOTE: This file is on its way to obsolescence. GnomeVFS is
#	taking over MIME-handling responsibility, which will eventually
#	be eliminated from gnome-libs (after GnomeVFS ships).
#
#	GnomeVFS uses "gnome-vfs.mime" and explicitly ignores this
#	file if found. Nautilus uses GnomeVFS.
#
#	While this file is still around, if you make corrections, please
#	also correct gnome-vfs.mime.
#
#
# Copryght (C) The Free Software Foundation, 1998, 1999
# Copyright (C) 2000 Eazel, Inc.
# http://www.fsf.org
#
# Mime types as provided by the GNOME libraries for GNOME.
#
# Applications can provide more mime types by installing other
# .mime files in the $share/mime-info directory.
#
# The format of this file is:
#
# mime-type
#         ext[,prio]:   list of extensions for this mime-type
#         regex[,prio]: a regular expression that matches the filename
#
# more than one ext: and regex: fields can be present.
#
# prio is the priority for the match, the default is 1.  This is required 
# to distinguish composed filenames, for example .gz has a priority of 1
# and .tar.gz has a priority of 2 (thus a file having the filename
# something.tar.gz will match the mime-type for tar.gz before the mime-type
# for .gz
#

application/andrew-inset
	ext: ez

application/mspowerpoint
	ext: ppt

application/msword
	ext: doc

application/octet-stream
	ext: bin

application/oda
	ext: oda

application/pdf
	ext: pdf

application/pgp
	ext: pgp

application/postscript
	ext: ps eps

application/rtf
	ext: rtf

application/vnd.ms-excel
	ext: xls

application/x-applix-spreadsheet
        ext: as

application/x-applix-word
        ext: aw

application/x-arj
	ext: arj

application/x-asp
        ext: asp

application/x-backup
        ext: bak BAK
	
application/x-bcpio
	ext: bcpio

application/x-bzip
	ext: bz2 bz

application/x-bzip-compressed-tar
	regex,2: tar\.bz2$

application/x-cgi
        ext: cgi

application/x-chess-pgn
	ext: pgn

application/x-class-file
	ext: class

application/x-compressed-tar
	regex,2: tar\.gz$
	ext: tgz

application/x-core-file
	regex: core

application/x-cpio
	ext: cpio

application/x-cpio-compressed
	regex: \.cpio\.gz$

application/x-deb
	ext: deb

application/x-dvi
	ext: dvi

application/x-e-theme
        ext: etheme ETHEME

application/x-font-afm
	ext: afm

application/x-font-bdf
	ext: bdf

application/x-font-linux-psf
	ext: psf

application/x-font-pcf
	ext: pcf

application/x-font-speedo
	ext: spd

application/x-font-ttf
	ext: ttf TTF

application/x-font-type1
	ext: pfa pfb

application/x-gnome-app-info
	ext: desktop

application/x-gnumeric
	ext: gnumeric

application/x-gtar
	ext: gtar

application/x-gzip
	ext: gz

application/x-hdf
	ext: hdf

application/x-kde-app-info
	ext: kdelnk

application/x-lha
	ext: lha

application/x-lhz
	ext: lhz

application/x-mif
	ext: mif

application/x-netcdf
	ext: cdf nc

application/x-object-file
	ext: o

application/x-ogg
        ext: ogg

application/x-php
        ext: php php3 php4

application/x-profile
        regex: gmon\.out$

application/x-reject
	ext: rej
	
application/x-rpm
	ext: rpm

application/x-shar
	ext: shar

application/x-shared-library
	ext: so

application/x-shared-library-la
        ext: la

application/x-smil
	ext: smil smi sml

application/x-staroffice-presentation
	ext: sdd

application/x-staroffice-spreadsheet
	ext: sdc

application/x-staroffice-words
	ext: sdw

application/x-sv4cpio
	ext: sv4cpio

application/x-sv4crc
	ext: sv4crc

application/x-tar
	ext: tar

application/x-theme
        ext: theme

application/x-troff-man-compressed
	regex: ([^0-9]|^[^\.]*)\.([1-9][a-z]?|n)\.g?[Zz]$

application/x-unix-archive
	ext: a sa

application/x-ustar
	ext: ustar

application/x-wais-source
	ext: src

application/x-zoo
	ext: zoo

application/zip
	ext: zip



audio/basic
	ext: snd

audio/x-aiff
	ext: aif aifc aiff

audio/x-it
        ext: it IT

audio/x-midi
	ext: midi mid

audio/x-mod
	ext: mod MOD

audio/x-mp3
	ext: mp3

audio/x-mpegurl
        ext: m3u

audio/x-pn-realaudio
	ext: rm ra ram

audio/x-s3m
	ext: S3M s3m

audio/x-stm
	ext: STM stm

audio/x-ulaw
	ext: au

audio/x-voc
	ext: voc

audio/x-wav
	ext: wav

audio/x-xi
	ext: xi

audio/x-xm
	ext: xm XM



image/cgm
	ext: cgm

image/gif
	ext: gif

image/ief
	ext: ief

image/jpeg
	ext: jpe jpeg jpg

image/png
	ext: png

image/svg
	ext: svg

image/tiff
	ext: tif tiff

image/vnd.dwg
        ext: dwg

image/vnd.dxf
        ext: dxf

image/x-3ds
        ext: 3ds

image/x-applix-graphic
        ext: ag

image/x-bmp
	ext: bmp

image/x-cmu-raster
	ext: ras

image/x-compressed-xcf
	regex,2: \.xcf\.gz$
	regex.2: \.xcf\.bz2$

image/x-ico
	ext: ico

image/x-iff
        ext: iff IFF

image/x-ilbm
        ext: ilbm ILBM

image/x-lwo
        ext: lwo lwob

image/x-lws
        ext: lws

image/x-portable-anymap
	ext: pnm

image/x-portable-bitmap
	ext: pbm

image/x-portable-graymap
	ext: pgm

image/x-portable-pixmap
	ext: ppm

image/x-psd
	ext: psd

image/x-rgb
	ext: rgb

image/x-tga
	ext: tga

image/x-xbitmap
	ext: xbm

image/x-xcf
	ext: xcf

image/x-xfig
        ext: fig

image/x-xpixmap
	ext: xpm

image/x-xwindowdump
	ext: xwd



message/x-gnu-rmail
	regex: ^RMAIL$



model/vrml
	ext: wrl




text/bib
	ext: bib

text/css
	ext: css

text/html
	ext: html htm HTML

text/mathml
	ext: mml

text/plain
	ext: asc txt TXT

text/richtext
	ext: rtx

text/sgml
	ext: sgml sgm

text/tab-separated-values
	ext: tsv

text/x-authors
	regex: AUTHORS
	
text/x-c
	ext: c h

text/x-c++
	ext: cc C cpp c++ hp H h++

text/x-comma-separated-values
	ext: csv

text/x-copying
	regex: COPYING	

text/x-credits
	regex: CREDITS

text/x-csh
	ext: csh

text/x-dcl
	ext: dcl

text/x-dsl
	ext: dsl

text/x-dtd
	ext: dtd

text/x-emacs-lisp
	ext: el

text/x-fortran
	ext: f

text/x-gtkrc
        regex: [.|]gtkrc

text/x-idl
        ext: idl

text/x-install
	regex: INSTALL

text/x-java
	ext: java

text/x-makefile
	regex: [Mm]akefile

text/x-perl
	ext: pl perl

text/x-python
	ext: py

text/x-readme
	regex: README.*

text/x-scheme
	ext: scm

text/x-setext
	ext: etx

text/x-sh
	ext: sh

text/x-sql
	ext: sql

text/x-tcl
	ext: tcl

text/x-tex
	ext: tex

text/x-texinfo
	ext: texi texinfo

text/x-troff
	ext: roff t tr

text/x-troff-man
	regex: (([^0-9]|^[^\.]*)\.([1-9][A-Za-z]*|n)|\.man)$
	ext: man

text/x-troff-me
	ext: me

text/x-troff-ms
	ext: ms

text/x-vcalendar
	ext: vcs vcf

text/x-vcard
	ext: gcrd

text/xml
	ext: xml

video/mpeg
	ext: mp2 mpe mpeg mpg

video/quicktime
	ext: mov qt

video/x-anim
        ext: anim5 anim7

video/x-flc
	ext: flc

video/x-fli
	ext: fli

video/x-msvideo
	ext: avi

video/x-sgi-movie
	ext: movie
