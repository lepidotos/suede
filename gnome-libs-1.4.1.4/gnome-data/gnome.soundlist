[__section_info__]
description=GNOME system events
description[ca]=Events de sistema de GNOME
description[da]=Systemhændelser i Gnome skrivebordsmiljøet
description[de]=Sound-Ereignisse für GNOME
description[el]=Ãåãïíüôá ôïõ óõóôŞìáôïò GNOME
description[es]=Eventos sistema de GNOME
description[et]=GNOME süsteemsed sündmused
description[fi]=GNOMEn järjestelmätapahtumat
description[fr]=Événements système de GNOME
description[gl]=Eventos de sistema de GNOME
description[it]=Eventi di sistema GNOME
description[ko]=±×³ğ ½Ã½ºÅÛ ÀÌº¥Æ®
description[lt]=GNOME sistemos ávykiai
description[no]=Systemhendelser for GNOME
description[pl]=Zdarzenia systemowe GNOME
description[pt]=Eventos de sistema GNOME
description[sk]=Systémové udalosti GNOME
description[tr]=Gnome sistem eylemleri
description[uk]=óÉÓÔÅÍÎ¦ ĞÏÄ¦§ GNOME
description[wa]=Evenmints sistinme di GNOME
description[zh_TW]=GNOME ¨t²Î¨Æ¥ó

[login]
file=login.wav
description=Login
description[ca]=Entrada al sistema
description[da]=Indlogning
description[de]=Anmeldung zur Sitzung
description[el]=¸íáñîç óõíåäñßáò
description[es]=Apertura de sesión
description[et]=Sisselogimine
description[fi]=Sisään kirjautuminen
description[fr]=Ouverture de session
description[gl]=Comezo de sesión
description[it]=Login
description[ko]=·Î±×ÀÎ
description[lt]=Prisijungimas
description[no]=Innlogging
description[pl]=Rozpoczêcie pracy
description[pt]=Início de Sessão
description[sk]=Prihlásenie
description[tr]=Giriş
description[uk]=÷È¦Ä Õ ÓÉÓÔÅÍÕ
description[wa]=Intreye è session
description[zh_TW]=µn¤J

[logout]
file=logout.wav
description=Logout
description[ca]=Eixida del sistema
description[da]=Udlogning
description[de]=Abmeldung von der Sitzung
description[el]=¸îïäïò áğü ôç óõíåäñßá
description[es]=Salida de sesión
description[et]=Väljalogimine
description[fi]=Ulos kirjautuminen
description[fr]=Fin de session
description[gl]=Fin de sesión
description[it]=Esci da GNOME
description[ko]=·Î±×¾Æ¿ô
description[lt]=Atsijungimas
description[no]=Utlogging
description[pl]=Zakoñczenie pracy
description[pt]=Fim de Sessão
description[sk]=Odhlásenie
description[tr]=Çıkış
description[uk]=úÁ×ÅÒÛÅÎÎÑ ÒÏÂÏÔÉ
description[wa]=Cwand li session est hûte
description[zh_TW]=µn¥X

[info]
file=info.wav
description=Informational Messages
description[ca]=Missatges informatius
description[da]=Informative beskeder
description[de]=Informationen
description[el]=Mçíıìáôá Ğëçñïöüñçóçò
description[es]=Mensajes informativos
description[et]=Teated
description[fi]=Tiedoteviestit
description[fr]=Messages d'information
description[gl]=Mensaxes de información
description[it]=Informazioni
description[ko]=¾Ë¸² ¸Ş¼¼Áö
description[lt]=Informacinës şinutës
description[no]=Informative meldinger
description[pl]=Komunikat informacyjny
description[pt]=Mensagens Informativas
description[sk]=Informácie
description[tr]=Bilgi iletileri
description[uk]=¶ÎÆÏÒÍÁÃ¦ÊÎ¦ ĞÏ×¦ÄÏÍÌÅÎÎÑ
description[wa]=Messaedjes d' informåcion
description[zh_TW]=¸ê®Æ°T®§

[warning]
file=warning.wav
description=Warning Messages
description[ca]=Missatges d'avís
description[da]=Advarsler
description[de]=Warnungen
description[el]=Ìçíıìáôá ĞñïåéäïğïéŞóåùí
description[es]=Mensajes de advertencia
description[et]=Hoiatused
description[fi]=Varoitusviestit
description[fr]=Messages d'avertissement
description[gl]=Mensaxes de aviso
description[it]=Avviso
description[ko]=°æ°í ¸Ş¼¼Áö
description[lt]=Áspëjamosios şinutës
description[no]=Advarsler
description[pl]=Komunikat ostrzegawczy
description[pt]=Avisos
description[sk]=Varovania
description[tr]=Uyarı iletileri
description[uk]=ğÏĞÅÒÅÄÖÕ×ÁÌØÎ¦ ĞÏ×¦ÄÏÍÌÅÎÎÑ
description[wa]=Messaedjes d' avertixhmint
description[zh_TW]=Äµ§i°T®§

[error]
file=error.wav
description=Error Messages
description[ca]=Missatges d'error
description[da]=Fejlmeldinger
description[de]=Fehler
description[el]=Ìçíıìáôá óöÜëìáôïò
description[es]=Mensajes de error
description[et]=Veateated
description[fi]=Virheviestit
description[fr]=Messages d'erreur
description[gl]=Mensaxes de erro
description[it]=Errore
description[ko]=¿À·ù ¸Ş¼¼Áö
description[lt]=Klaidø praneğimai
description[no]=Feilmeldinger
description[pl]=Komunikat b³êdu
description[pt]=Mensagens de Erro
description[sk]=Chyby
description[tr]=Hata iletileri
description[uk]=ğÏ×¦ÄÏÍÌÅÎÎÑ ĞÒÏ ĞÏÍÉÌËÕ
description[wa]=Messaedjes d' aroke
description[zh_TW]=¿ù»~°T®§

[question]
file=question.wav
description=Question Dialogs
description[ca]=Diàlegs interrogatius
description[da]=Spørgsmålsdialoger
description[de]=Fragen
description[el]=ÄéÜëïãïé ìå åñùôŞóåéò
description[es]=Diálogos con preguntas
description[et]=Küsimused
description[fi]=Kysymysvalintaikkunat
description[fr]=Dialogues avec des questions
description[gl]=Diálogos con preguntas
description[it]=Domanda
description[ko]=¹°À½ ´ëÈ­»óÀÚ
description[lt]=Klausimø dialogai
description[no]=Spørsmålsdialoger
description[pl]=Komunikat z zapytaniem
description[pt]=Diálogos com questões
description[sk]=Otázky
description[tr]=Sorgulama iletileri
description[uk]=ä¦ÁÌÏÇÉ ¦Ú ÚÁĞÉÔÁÎÎÑÍÉ
description[wa]=Purneas di dialogue avou des kesses
description[zh_TW]=¸ß°İ¹ï¸Üµ¡

[generic]
file=generic.wav
description=Miscellaneous Messages
description[ca]=Missatges diversos
description[da]=Diverse beskeder
description[de]=Sonstige Meldungen
description[el]=ÄéÜöïñá ìçíıìáôá
description[es]=Mensajes misceláneos
description[et]=Muud teated
description[fi]=Sekalaiset viestit
description[fr]=Autres messages
description[gl]=Outras mensaxes
description[it]=Evento generico
description[ko]=±× ¿Ü ¸Ş¼¼Áö
description[lt]=Ávairios şinutës
description[no]=Forskjellige beskjeder
description[pl]=Ró¿ne komunikaty
description[pt]=Outras mensagens
description[sk]=Rôzne správy
description[tr]=Diğer iletiler
description[uk]=ò¦ÚÎ¦ ĞÏ×¦ÄÏÍÌÅÎÎÑ
description[wa]=Ôtes messaedjes
description[zh_TW]=¨ä¥¦°T®§
