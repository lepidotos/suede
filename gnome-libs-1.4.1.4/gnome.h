#ifndef GNOME_H
#define GNOME_H

#if defined(GNOME_INTERFACE_VERSION)
#if GNOME_INTERFACE_VERSION > 1
#error "You are picking up an old header file because of a bad include path."
#error "Please make sure `gnome-config --cflags ...` output is before"
#error "any other -I options in your compilation flags."
#endif
#endif

#include "gnomesupport.h"

#include <gtk/gtk.h>
#include <gdk_imlib.h>
#include "libgnome/libgnome.h"
#include "libgnomeui/libgnomeui.h"
#endif
