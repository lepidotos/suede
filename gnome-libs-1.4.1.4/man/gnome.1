.\"
.\" gnome manual page
.\" (C) 1999, 2000 Miguel de Icaza (miguel@gnu.org).
.\" 
.\" This manual page is covered by the terms of the GNU General
.\" Public License.  
.\"
.TH GNOME 1 "GNOME 1.2" 
.SH NAME
GNOME \- The GNU Network Object Model Environment
.SH SYNOPSIS
.PP
.B gnome-session
.SH DESCRIPTION
.PP
The 
.B gnome-session
program launches and initializes the GNOME desktop environment.  This
program is usually executed from your X initialization file.  If
your system is configured to use
.B gdm
(the GNOME display and login manager) you can start your GNOME session
by selecting the Gnome login profile.
.PP
If the special 
.B WINDOW_MANAGER
environment variable is set, the gnome-session system will use that as
the session window manager.  Otherwise it will default to your
system's configured window manager.
.PP
GNOME is a collection of libraries and applications.  A collection of
these form the GNOME Desktop: an easy to use, yet powerful desktop
environment for Unix systems.  You can find up to date information
about GNOME in http://www.gnome.org.  You can find more information
about the GNU project in http://www.gnu.org.
.PP
From a user's point of view, the GNOME desktop consists of a desktop
metaphor, a file manager and an easy way to launch applications
installed on the system.  Various desktop tools are provided with the
GNOME desktop to take advantage of a computer system.  
.PP
GNOME's desktop metaphor allows the desktop to be used as a place to
temporarily storing files, shortcuts to programs and documents.  Drag
and drop is an important part of the system; we have tried to make the
system as intuitive as possible.
.PP
The session management in GNOME will automatically restore all of the
applications you were running when you log in into the system again.
With session managed applications, the user can turn off or logout
from the system and when he logs in again, he will see the same
desktop he had before.
.PP
GNOME supports themes that allow users to change the skin of an
application: the look of applications in the GNOME desktop can be
configured to look in the way that more pleases the user: it is just a
few mouse-clicks away.  You can choose from a wide range of GTK
themes.  A web site has been devoted to this: http://gtk.themes.org
.PP
.SH ARCHITECTURE
The GNOME architecture addresses a number of problems and missing
features found on Unix systems and it uses a number of components to
achieve this:
.PP
.SS glib
This is the foundation library that provides portability functions, a
collection of reusable abtract types for C programmers and a main loop
abstraction.  For more information see http://www.gtk.org/rdp/glib/book1.html
.SS ORBit
This is the CORBA implementation used in GNOME.  CORBA provides basic
RPC functionality and it is the foundation for the component model and
the compound document and document model systems.  For more
information see http://www.labs.redhat.com/orbit.
.SS GTK+
This is the GUI toolkit used by GNOME.  It works on Unix and Win32
systems and other ports are being worked on to lighter windowing
systems.  You can find more information on http://www.gtk.org/
.SS gtk-engines
The GTK+ toolkit has support for changing the apperance of application
by providing support for themes and theme engines.  See
http://gtk.themes.org for a collection of readily-available themes. 
.SS Imlib
The graphics library used to load, save, manipulate and render images
in GNOME applications.  It includes routines to do fast drawing and
use a limited set of colors from low-end displays.  We expect this
library to be replaced soon with the more modern libart.
.SS libart_lgpl
An imaging library used for implementing various high-quality imaging
components in GNOME.
.SS gnome-libs
These libraries are the core libraries that provide the uniformity of
the applications.  They are divided in five: libgnome (for non-GUI
dependant code), libgnomeui (for GUI dependant code), zvt (the
xterm terminal emulator), gtk-xmhtml (an HTML rendering engine) and
libgnorba that implements the CORBA object activation and registry.
.SS libglade
This library enables programmers to create their interfaces using the
Glade GUI desginer and loading at runtime the user interfaces.
.SS gnome-print
The GNOME printing architecture implements a Postscript imaging model
with two extensions: alpha transparency and anti-aliasing (all of this
is done by using the libart_lgpl imaging library.  
.SS gnome-xml
This library provides GNOME application with an API to load, parse and
walk an XML file.  
.SS Docbook
GNOME documentation is written in the Docbook SGML DTD.  You can find
more about this at
http://nis-www.lanl.gov/~rosalia/mydocs/docbook-intro.html
.SS GNU gettext
GNOME uses the GNU gettext to allow applications to be localized for
various countries and languages.
.SS Bonobo
Bonobo is the GNOME architecture for creating reusable software
components and compound documents.  It was designed and implemented to
address the needs and problems of the free-software community for
developing large-scale applications.
.PP
More information can be found at
http://www.helixcode.com/tech/bonobo.php3
.PP
GNOME is window manager independant.  This means that the GNOME
desktop and the GNOME tools will work with any window manager.  Window
manager can optionally provide a number of features that will make the
user's desktop a more pleasant experience.  The GNOME window manager
hint spec is available at: http://www.gnome.org/devel/gnomewm/book1.html
.SH HISTORY
There were two projects that lead to the creation of origins of what
became the GNOME project: the libapp project and the old-GNOME
project.  The former was a project to provide standard
workstation-like services to applications.  The old-GNOME project was
intended to provide a component model for Unix systems.  These were
projects some of us had discussed but never actually implemented.
.PP
Enter KDE,  a project that wanted to make Unix usable as a desktop
machine.  Sadly they chose the proprietary and non-free toolkit Qt
as the foundation for their work.  It was a giant step backwards in
terms of software freedom[1].
.PP
In response, the GNOME project was started later to create a
completely free desktop environment, and various early ideas were
reused.
.PP
Early talks about the creation of GNOME involved some recognized free
software leaders: Erik Troan and Mark Ewing of Red Hat software,
Richard Stallman of the Free Software Foundation, and Peter Mattis and
Spencer Kimball of the GIMP project.  We launched the project after
considering the various alternatives that could be tried.
.PP
The original call for developers, which included the team of
programmers working on the GNU Image Manipulation Program (GIMP
[GIMP]), the Guile mailing list and the free software mailing lists.
This is important because the mix of people that were part of the
original GNOME team had a good background on free software issues,
graphics and language design.
.PP
Red Hat created the Advanced Development Laboratories division on
January 1998 (http://www.labs.redhat.com).  RHAD labs was initially
created to help out in the development of the GNOME project.
.PP
We made releases of the GNOME source base since the beginning of the
project.  During the development of GNOME, the group has produced a
number of libraries and components that are useful to provide
integration, and consistency troughout the system.
.PP
GNOME 1.0 was released after eighteen months of development in March
1999.  Updates and fixes are continously released; At the time of this
writing, the GNOME 1.0 series is at version 1.0.5.
.PP
GNOME 1.0 marks the contract between GNOME developers and the user
base to provide a stable API on top of which new applications can be
developed.  Software developers will be able to take advantage of all
the functions available in the library, and they can be sure that
their applications will continue to work in the future.
.PP
In May, 1999, International GNOME support was launched: a company that
offers contractual support for the GNOME system founded by Nat
Friedman and Miguel de Icaza.
.PP
In October, 1999 an updated version of GNOME codenamed "October GNOME" was
released with many bug fixes and improvements.  This new version of
GNOME also included Glade and libGlade as part of the platform
.PP
In October 1999, GNOME Support became Helix Code,
Inc. (http://www.helixcode.com) and started work on Evolution (an
integrated groupware solution) and Helix GNOME (a continous updated
distribution of GNOME for various operating systems).
.PP
In November 1999, Eazel was introduced to the GNOME community
(http://www.eazel.com) founded by Andy Hertzfeld, Bart Decrem and Mike
Boich to provide a new desktop for GNOME: the Nautilus project.
.PP
Also in November, the Bonobo component system started to become used
in the GNOME project, and it became the foundation for various of the
most advanced GNOME projects.
.PP
In March 2000, Mathieu Lacage organized the "GNOME Users and
Developers European Conference" (http://www.guadec.enst.fr) in the
Telecom, Paris school in Paris, France.  More than a hundred GNOME
hackers got together to discuss the state of GNOME and its future.
.PP
In March 2000, The GNOME Steering Committee was created to overwsee
the development and deployment of GNOME 2.0
.PP
In May 2000, GNOME 1.2 codenamed "Bongo GNOME" was released to the
public.
.PP
.SH MAILING LISTS
There are various mailing lists used by the GNOME project to
coordinate the development of GNOME, you can subscribe to these lists
by sending mail to the <listname>-request@domain address and put in
the body of your message the word "subscribe". 
.PP
.SS gnome-announce-list@gnome.org
Where general announcements about the GNOME system are done.  A good
way of staying in touch with the developments of the system
.PP
.SS gnome-list@gnome.org
General discussion of the GNOME system.
.PP
.SS gnome-devel-list@gnome.org
Discussions on the development of the GNOME system and on writing
GNOME applications.
.PP
.SS gnome-gui-list@gnome.org
Discussion about user interface improvements for the GNOME system. 
.PP
.SS gnome-components-list@gnome.org
Discussions about Bonobo: the component and compound document
architecture of GNOME.
.PP
.SS cvs-commits-list@gnome.org
Used to keep track of changes to the GNOME CVS source code repository.
.PP
There are many other lists that discuss specific parts of the project,
for a complete list, check http://www.gnome.org/mailing-lists
.PP
.SH BUGS
.PP
To report bugs or suggestions you would like to see in the GNOME
system, please use the command
.B gnome-bug
to send us information about the problem you are experimenting, or go
directly to our bug tracking system on the Web at
http://bugs.gnome.org
.SH AUTHOR
.PP
GNOME has been developed by a large number of free software
programmers, users and enthusiasts on the Internet.  The
.B guname
program lists some of the contributors to the system.
.PP
This manual page has been written by Miguel de Icaza (miguel@gnu.org)
