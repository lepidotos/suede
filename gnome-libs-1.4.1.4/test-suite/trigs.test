the_test=trigs
. $srcdir/runtest.sh
