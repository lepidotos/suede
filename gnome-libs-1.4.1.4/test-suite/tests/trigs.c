#include <gnome.h>
#include <glib.h>
#include <stdio.h>

int global_cntr = 0;

static void sample_trigger_function(char *msg, char *level, char *supinfo[])
{
	int i;
	for(i = 0; supinfo[i]; i++)
		g_print("%s ", supinfo[i]);
	g_print("[%s] %s\n", level, msg);
	fflush(stdout);
	
	global_cntr++;
}

int main(int argc, char *argv[])
{
	struct _GnomeTrigger trig;
	gchar *fn;

	trig.type = GTRIG_FUNCTION;
	trig.u.function = sample_trigger_function;
	trig.level = NULL;

	gnomelib_init("trigs", "0.0");
	fn = g_strconcat(getenv("srcdir"), "/tests/trigs.in", NULL);
	gnome_triggers_init();
	gnome_triggers_add_trigger(&trig, "test", "one", NULL);
	gnome_triggers_do("Test of direct hit", "warning", "test", "one", NULL);
	fflush(stdout);
	g_print("Global counter is now %d\n", global_cntr);
	fflush(stdout);
	gnome_triggers_do("Test of indirect hit", "warning", "test", "one", "two", "three", NULL);
	fflush(stdout);
	g_print("Global counter is now %d\n", global_cntr);
	fflush(stdout);
	return 0;
}
