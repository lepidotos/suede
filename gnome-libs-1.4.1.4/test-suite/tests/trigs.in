warning test:one:two command /bin/echo
