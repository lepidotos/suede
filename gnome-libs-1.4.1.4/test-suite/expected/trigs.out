test one [warning] Test of direct hit
Global counter is now 1
test one two three [warning] Test of indirect hit
Global counter is now 2
Exit code 0
